src/store/plugins/reproject-everything-on-projection-change.plugin.js

```diff
@@ -77,7 +76,7 @@ const reprojectEverythingOnProjectionChangePlugin = (store) => {
                     store.dispatch('setSelectedFeatures', reprojectedSelectedFeatures)
                 })
 
-                if (newProjection.epsg === LV95.epsg) {
+                if (newProjection.isSwissProjection) {
                     store.dispatch(
                         'setZoom',
                         translateMercatorZoomToSwisstopoPyramidZoom(state.position.zoom)
```

Comment:
Here it may be wise to have a new function from the zoom utilities called `translateZoom(fromProjection, toProjection, zoom)` and that this method will call the correct `translate...` method or throw and error if we don't support the translation.
So for the zoom here we would have a very short code 

```javascript
store.dispatch('setZoom', translateZoom(oldProjection.epsg, newProjection.epsg, state.position.zoom))
```